TileMapLayer::navmesh_parse_init() missing NAVIGATION_2D_DISABLED guard in register_types.cpp

### Tested versions

- Reproducible in: 4.7.1.stable

### System information

Windows 11 IoT Enterprise LTSC - Godot v4.7.1.stable - Building export templates with SCons + GDBuild

### Issue description

When building Godot 4.7.1 export templates using SCons (via GDBuild), compilation intermittently fails with the following error:

```text
ERROR: modules\tilemap\register_types.cpp: In function 'void initialize_tilemap_module(ModuleInitializationLevel)':
modules\tilemap\register_types.cpp:58:31: error: 'navmesh_parse_init' is not a member of 'TileMapLayer'
   58 |                 TileMapLayer::navmesh_parse_init();

      |                               ^~~~~~~~~~~~~~~~~~
```

The call to TileMapLayer::navmesh_parse_init() at line 58 of modules/tilemap/register_types.cpp is missing the #ifndef NAVIGATION_2D_DISABLED preprocessor guard. When Navigation2D is disabled in the build profile, this method does not exist, causing a compile-time failure.
Expected behavior: The engine should compile successfully regardless of whether Navigation2D is enabled or disabled.

English is not my first language. This issue was translated with AI assistance to ensure technical accuracy. Please let me know if any clarification is needed.

### Steps to reproduce

1. Set up a custom build profile with Navigation2D disabled (or use GDBuild with equivalent configuration)
2. Build Godot 4.7.1 export templates using SCons
3. Compilation fails at `modules/tilemap/register_types.cpp:58

### Minimal reproduction project (MRP)

N/A – This is a compile-time issue in the engine source code (`modules/tilemap/register_types.cpp`). No runtime project is needed.

--- a/modules/tilemap/register_types.cpp
+++ b/modules/tilemap/register_types.cpp
@@ -53,9 +53,14 @@ void initialize_tilemap_module(ModuleInitializationLevel p_level) {
 		GDREGISTER_CLASS(TileMapLayer);
 #ifndef DISABLE_DEPRECATED
 		GDREGISTER_CLASS(TileMap);
+#endif
+
+#ifndef NAVIGATION_2D_DISABLED
+		TileMapLayer::navmesh_parse_init();
+#ifndef DISABLE_DEPRECATED
 		TileMap::navmesh_parse_init();
 #endif
-		TileMapLayer::navmesh_parse_init();
+#endif // NAVIGATION_2D_DISABLED
 	}
 #ifdef TOOLS_ENABLED
 	if (p_level == MODULE_INITIALIZATION_LEVEL_EDITOR) {

